### PR TITLE
ENH center all umap embeddings + migrate to github actions

### PR DESCRIPTION
**Pull request by gwarmstrong**
This PR contains:
* all embeddings are centered automatically
* removal of q2 access to center command
* small wording change in readme

### CODE REVIEW

**Comment by antgonza** [comment]
Note that we are not longer using travis as a CI due to credits, you need to move to github actions ...

**Comment by gwarmstrong** [comment]
> Note that we are not longer using travis as a CI due to credits, you need to move to github actions ...

Yes. I had set up the Travis for this before the migration... Working on the GitHub actions as we speak.

**Comment by gwarmstrong** [comment]
@antgonza CI should be good now

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+name: q2-umap CI
+
+on: [ push, pull_request ]
```
Comment (antgonza) [inline .github/workflows/python-package.yml:3]: Do you really need these 2? The tests are basically the same for both, no?

File: README.md [hunk 2]
```diff
@@ -52,15 +52,4 @@ qiime emperor plot \
     --m-metadata-file sample-metadata.tsv \
     --o-visualization umap-emperor.qzv
     
-# optionally
-qiime umap center \
-  --i-embedding jaccard_umap.qza \
-  --o-centered-embedding centered_jaccard.qza
-  
-qiime emperor plot \
```
Comment (antgonza) [inline README.md:60]: Why remove the emperor example?

**Review by antgonza — COMMENTED** [review]
Looks good, thank you. A few minor comments + I'm actually surprised that the tests for embed didn't change - as now the results are being centered, any idea why?

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (gwarmstrong) [reply .github/workflows/python-package.yml:3]: I frequently push to branches before opening PR's and I find it nice when CI is run on branches without an open PR.

File: README.md (hunk 2 above)
Comment (gwarmstrong) [reply README.md:60]: this only removes the emperor example on the manually centered data. See line 50 of this file for another emperor visualization following call to `qiime umap embed`

**Comment by antgonza** [comment]
Just to be sure ... so it's fine that the tests didn't change after now embed uses center, right?

**Comment by gwarmstrong** [comment]
> Just to be sure ... so it's fine that the tests didn't change after now embed uses center, right?

I have not checked values of the output of UMAP, since would be intractable to solve by hand, and not really what our package guarantees. Instead, I have checked that surface level aspects are correct, e.g., shape, labels, values are centered at 0 with largest variance along PC1, etc.

I can add a test for values pre-computed with a set random seed if you think that would be good, but should only really be affected if there are major backwards-incompatible changes with the `umap` package itself.

**Comment by antgonza** [comment]
Thank you for the clarification; I guess that check should be in the umap package ...

**Review by antgonza — APPROVED** [review]

**Pull request merged**